src/wasm/wat-lexer.cpp

```diff
@@ -781,6 +781,136 @@ std::optional<LexResult> keyword(std::string_view in) {
 
 } // anonymous namespace
 
+std::optional<uint64_t> Token::getU64() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    if (tok->sign == NoSign) {
+      return tok->n;
+    }
+  }
+  return {};
+}
+
+std::optional<int64_t> Token::getS64() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    // Integers without an explicit sign still need to be in signed range.
+    if (tok->sign != NoSign || tok->n < (1ull << 63)) {
+      return int64_t(tok->n);
+    }
+    // TODO: Add error production for out-of-bounds integers.
+  }
+  return {};
+}
+
+std::optional<uint64_t> Token::getI64() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    return tok->n;
+  }
+  return {};
+}
+
+std::optional<uint32_t> Token::getU32() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    if (tok->sign == NoSign && tok->n < (1ull << 32)) {
+      return int32_t(tok->n);
+    }
+    // TODO: Add error production for out-of-bounds integers.
+  }
+  return {};
+}
+
+std::optional<int32_t> Token::getS32() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    if (tok->n < (1ull << 31) ||
+        (tok->sign == Neg && -(1ull << 31) <= tok->n)) {
+      return int32_t(tok->n);
+    }
+    // TODO: Add error production for out-of-bounds integers.
+  }
+  return {};
+}
+
+std::optional<uint32_t> Token::getI32() const {
+  if (auto n = getU32()) {
+    return *n;
+  }
+  if (auto n = getS32()) {
+    return uint32_t(*n);
+  }
+  return {};
+}
+
+std::optional<double> Token::getF64() const {
+  constexpr int signif = 52;
```

Comment:
What does "if" mean in "signif"? Or am I reading this name wrong?

Reply:
`signif` is short for "significand" and is the name used for this in the spec: https://webassembly.github.io/spec/core/syntax/values.html#aux-significand.

Reply:
Ah, thanks...